synapse/rest/client/v2_alpha/account.py

```diff
@@ -295,7 +296,7 @@ def on_GET(self, request, medium):
             )
             request.setResponseCode(e.code)
 
-        request.write(html.encode('utf-8'))
+        request.write(ensure_binary(html))
```

Comment:
sorry, but this still doesn't look right. the bug here is that `html` is having an identity crisis over whether it contains unicode or bytes. Presumably that's happening because the things you're assigning to html are having similar identity crises.

The solution here isn't to go round sticking plasters pver that problem every time we spit out some html: rather it is to make sure that things are consistently unicode or bytes.

In other words: this needs doing earlier on. Presumably the problem is where `email_password_reset_success_html_content` is read from disk. go and fix it there instead.

Reply:
Ah, right, I understand now. Roger that, I'll look into it.